cpp/wrappers/WKTJsiArrayWrapper.h

```diff
@@ -179,25 +225,63 @@ class JsiArrayWrapper : public JsiHostObject,
     std::unique_lock lock(_readWriteMutex);
 
     auto callbackFn = arguments[0].asObject(runtime).asFunction(runtime);
+    
+    std::vector<jsi::Value> args(3);
+    args[2] = thisValue.asObject(runtime);
+    
     for (size_t i = 0; i < _array.size(); i++) {
-      auto arg = JsiWrapper::unwrap(runtime, _array.at(i));
-      auto retVal = callFunction(runtime, callbackFn, thisValue, &arg, 1);
-      if (retVal.getBool() == false) {
+      args[0] = _array.at(i)->unwrap(runtime);
+      args[1] = jsi::Value(static_cast<double>(i));
+      
+      auto retVal = callFunction(runtime, callbackFn, thisValue,
+                                 static_cast<const jsi::Value *>(args.data()), 3);
+
+      if (evaluateAsBoolean(runtime, retVal) == false) {
         return false;
       }
     }
     return true;
   };
+                          
+  JSI_HOST_FUNCTION(some) {
+    std::unique_lock lock(_readWriteMutex);
+
+    auto callbackFn = arguments[0].asObject(runtime).asFunction(runtime);
+    
+    std::vector<jsi::Value> args(3);
+    args[2] = thisValue.asObject(runtime);
+    
+    for (size_t i = 0; i < _array.size(); i++) {
+      args[0] = _array.at(i)->unwrap(runtime);
+      args[1] = jsi::Value(static_cast<double>(i));
+      
+      auto retVal = callFunction(runtime, callbackFn, thisValue,
+                                 static_cast<const jsi::Value *>(args.data()), 3);
```

Comment:
I guess this is more of a stylistic question, but Marc taught me to not do too much on one line and separate different instructions out:

```cpp
const jsi::Value* jsiArgs = static_cast<const jsi::Value *>(args.data());
auto retVal = callFunction(runtime, callbackFn, thisValue, jsiArgs, 3);
```

His arguments are that its easier to read + better for debugging / digesting stacks in case something crashes.

Just wanted to bring that up, but ultimately I don't mind 😄

Reply:
I'm kind of leaning to the opposite for things that we repeat a lot like the argument casting - its maybe hard to read the first time, but then you are happy to not see it anymore :) :) But that's just me and my rigid opinion :)